web-ng/src/app/components/layer-dialog/layer-dialog.component.ts

```diff
@@ -116,10 +116,7 @@ export class LayerDialogComponent implements OnDestroy {
 
   init(projectId: string, createLayer: boolean, layer?: Layer) {
     this.projectId = projectId;
-    if (!createLayer && !layer) {
-      throw Error('Layer not found');
-    }
-    if (createLayer) {
+    if ((!createLayer && !layer) || createLayer) {
```

Comment:
To avoid nesting the entire body of the function of the body in a block, you can use short circuiting (i.e. return instead).

Also, is there a way to make this expression more self explanatory? Are any of these possible states actually errors that we'd like to log?